Sources/SwiftFusion/Geometry/Pose2.swift

```diff
@@ -114,106 +192,78 @@ public struct Pose2: Differentiable, TangentStandardBasis, KeyPathIterable {
   }
 
   /// The derivative of `rot` satisfiying convention (2) described above.
-  @derivative(of: rot)
+  @derivative(of: Pose2Coordinate.rot)
   @usableFromInline
   func vjpRot() -> (value: Rot2, pullback: (Vector1) -> Vector3) {
     // This pullback is similar to the pullback in `vjpT`, but for the second column of the
     // matrix. See the comment in `vjpT` for more information.
     return (rot, { Self.tangentVector(w: $0, v: Vector2.zero) })
   }
+}
 
-  /// Twist coordinates, with the first component for rotation and the remaining components for
-  /// translation.
-  ///
-  /// See the documentation comment on the declaration of `Pose2` for more information.
-  public typealias TangentVector = Vector3
-
-  /// Creates a tangent vector given rotation (`w`) and translation (`v`) components.
-  public static func tangentVector(w: Vector1, v: Vector2) -> Vector3 {
-    Vector3(w.x, v.x, v.y)
+// MARK: Coordinate Operators
+public extension Pose2Coordinate {
+  /// Product of two rotations.
+  @differentiable
+  static func * (lhs: Pose2Coordinate, rhs: Pose2Coordinate) -> Pose2Coordinate {
+    Pose2Coordinate(lhs.rot * rhs.rot, lhs.t + lhs.rot * rhs.t)
   }
 
-  /// Returns the rotation (`w`) and translation (`v`) components of `tangentVector`.
-  public static func decomposed(tangentVector: Vector3) -> (w: Vector1, v: Vector2) {
-    (Vector1(tangentVector.x), Vector2(tangentVector.y, tangentVector.z))
+  /// Inverse of the rotation.
+  @differentiable
+  func inverse() -> Pose2Coordinate {
+    Pose2Coordinate(self.rot.inverse(), self.rot.unrotate(-self.t))
   }
+}
 
-  /// Moves `self` by `exp(hat(direction))`.
+extension Pose2Coordinate: ManifoldCoordinate {
   public mutating func move(along direction: Vector3) {
-    // TODO: This should be the real exponential map.
     let (w, v) = Self.decomposed(tangentVector: direction)
     self.tStorage.move(along: rot.rotate(v))
     self.rotStorage.move(along: w)
   }
-
-  /// Storage for the pose's translation.
+  
+  /// p * Exp(q)
+  @differentiable(wrt: local)
+  public func global(_ local: Vector3) -> Self {
+    self * Pose2Coordinate(Rot2(local.x), Vector2(local.y, local.z))
```

Comment:
The "Exp(q)" here should actually be the last formula in section 4.1 of https://github.com/borglab/gtsam/blob/develop/doc/LieGroups.pdf , because that is the formula for the exp on twist coordinates.

Here's an attempt at it, which I haven't compiled or tested, but if you run the tests it will probably reveal any problems:
```
let (w, v) = Pose2Coordinate.decomposed(tangentVector: local)
let vPerpOverW = Vector2(-v.y / w, v.x / w)
let expW = Rot2(w)
let expTwist = Pose2(expW, vPerpOverW - expW.rotate(vPerpOverW))
return self * expTwist
```

(the fact that we have to calculate the derivative of this thing any time we calculate the derivative of a function on Pose2 raises interesting questions about efficiency, but it's probably best to get something nice and simple working before worrying about that...)

Reply:
Now that I wrote that out, I notice that it does not work well for `w` near `0`, and it completely fails for `w == 0`. So for this approach to work, the implementation will need to be changed to be one that works better for `w` near `0`.

Reply:
Yep there is a near-zero approximation for that :)